Adding support from custom search providers

nmitsou opened the pull request:
A solution for #538 and #553 
The user is able to add, delete, select and reset custom search providers.

I use 3 StringSets in preferences:
- deleting-search-providers-names to store the providers to be delete
- available-search-providers to store all search providers in the format <name|url>
- selected-search-provider-names to store the names of the selected providers

When a user adds or deletes a provider, I have to remove the 2 multipreferences and add them again so that they are up to date.

Known issues: 
- When adding a search provider, the text might not be visible depending on the selected theme. 

This PR might increase the size of the app since I use 11 strings

Tested on android 7.1.1

nmitsou commented:
![screenshot_20170531-010330](https://cloud.githubusercontent.com/assets/7504889/26608694/3db207b0-459d-11e7-8f43-04cd223ffd2f.png)
![screenshot_20170531-010338](https://cloud.githubusercontent.com/assets/7504889/26608695/3dbf893a-459d-11e7-89b7-fc4a5793950f.png)
![screenshot_20170531-010355](https://cloud.githubusercontent.com/assets/7504889/26608697/3dc8e9d0-459d-11e7-81f6-4f7d964dd0c8.png)
![screenshot_20170531-010409](https://cloud.githubusercontent.com/assets/7504889/26608696/3dc63f78-459d-11e7-8fec-555363d46fb7.png)

smilecs commented on app/src/main/java/fr/neamar/kiss/dataprovider/SearchProvider.java:
```diff
@@ -14,44 +15,53 @@
-        searchProviderUrls.put("Wikipedia", "https://en.wikipedia.org/wiki/");
-        searchProviderUrls.put("Yahoo", "https://search.yahoo.com/search?p=");
+        defaultSearchProviders.add("Bing|https://www.bing.com/search?q=");
+        defaultSearchProviders.add("DuckDuckGo|https://duckduckgo.com/?q=");
```
why not define these url strings in a constants file or strings.xml though this practice was already there but we can improve from that

nmitsou commented:
I made some changes on this PR.
- Moved the default search providers in the xml file as suggested by @smilecs 
- Moved loading of search providers in the loader section

I still don't know how to make the dialog respect the theme colors...

Neamar commented:
@nmitsou if you fix the conflicts, just ping me and I'll merge ASAP before it gets into conflict again :)

nmitsou commented:
I fixed the conflicts. It should be fine now.
However, the dialog for adding new provider doesn't respect the theme colors... On white theme, the inserted text is not visible since it is also white.

Neamar commented:
Perfect!

Yes, there is still this bug on Light theme, but for now let's merge this. Thanks!
